src/Collateral.hs

```diff
@@ -78,34 +78,42 @@ mkValidator contractInfo@ContractInfo{..} dat interestPayDate ctx = validate
     getInterestAmnt v = assetClassValueOf v (interest dat)
 
     validateDebtAmnt :: TxOut -> Bool
-    validateDebtAmnt txo = getLoanAmnt (txOutValue txo) >= loanamnt dat
+    validateDebtAmnt txo = getLoanAmnt (txOutValue txo) >= loanAmnt dat
 
     interestPercentage :: Integer
-    interestPercentage = case (lendDate dat + repayinterval dat) < interestPayDate of
+    interestPercentage = case (lendDate dat + repayInterval dat) < interestPayDate of
```

Comment:
I'm not sure if this was communicated already, but would it be possible to use `loanDuration` instead? It's a much better name IMHO.

```suggestion
    interestPercentage = case (lendDate dat + loanDuration dat) < interestPayDate of
```